src/test/objectstore/allocsim/ops_scraper.py

```diff
@@ -0,0 +1,303 @@
+    if res.returncode != 0:
+        logger.error(f'error executing "{command}": \n{res.stderr}\n')
+    if no_out:
+        return None
+    return json.loads(res.stdout)
+
+def osd_ls() -> List[str]:
+    return run_ceph_command(['osd', 'ls'])
+
+def parse_osd_op(description_literal: str, initiated_at: datetime.datetime):
+    prev = 0
+    cursor = description_literal.find(' ', 0)
+    reqid = description_literal[prev:cursor]
+
+    prev = cursor
+    cursor = description_literal.find(' ', cursor+1)
+    pgid = description_literal[prev+1:cursor]
+
+    prev = cursor
+    cursor = description_literal.find(' ', cursor+1)
+    object_name = description_literal[prev+1:cursor]
+    object_name_split = object_name.split(':')
+    if not Env.args().nohash :
+        if len(object_name_split) < 2:
+            object_name = hashlib.md5(object_name.encode()).hexdigest()
+        else:
+            object_name_split[-2] = hashlib.md5(object_name_split[-2].encode()).hexdigest()
+            object_name = ':'.join(object_name_split)
+
+    prev = cursor
+    cursor = description_literal.find(']', cursor+1)
+    operations = description_literal[prev+2:cursor].split(',')
+    for i in range(len(operations)):
+        operations[i] = operations[i].split(' ')
+
+    return OpDescription(reqid, pgid, object_name, operations, initiated_at)
+
+class ProcessInfo:
+    def __init__(self, process_time: int, command_time: int, processed_info: str, ops_count: int,
+                 new_ops: int, oldest_operation: Optional[OpDescription], 
+                 capture_period_start: int, capture_period_end: int):
+        self.process_time = process_time
+        self.command_time = command_time
+        self.processed_info = processed_info
+        self.ops_count = ops_count
+        self.new_ops = new_ops
+        self.oldest_operation = oldest_operation
+        self.capture_period_start = capture_period_start
+        self.capture_period_end = capture_period_end
+
+def _to_timestamp(date: datetime.datetime) -> int:
+    return date.astimezone().timestamp()
+
+def process_osd(osd_id) -> ProcessInfo:
+    if osd_id not in Env.store():
+        Env.store()[osd_id] = {}
+        Env.store()[osd_id]['last_op'] = datetime.datetime.fromtimestamp(0, tz=datetime.timezone.utc)
+
+
+    logger.info(f'collecting from osd.{osd_id}')
+    command_time_start = time.time()
+    historic: Dict[str, Any] = run_ceph_command(['tell', f'osd.{osd_id}', 'dump_historic_ops'])
+    command_time = time.time() - command_time_start
+
+    osd_processing_start = time.time()
+
+    operations_str = 'op_type | offset/extent | name | pgid | initiated_at | who\n'
+    operations_str += '-------------------------------------------------------\n'
+    new_ops = 0 
+    oldest_operation = None
+    capture_period_start = _to_timestamp(Env.store()[osd_id]['last_op'])
+    for op in historic['ops']:
+        initiated_at = datetime.datetime.fromisoformat(op['initiated_at'])
```

Comment:
Traceback (most recent call last):
  File "../src/test/objectstore/allocsim/ops_scraper.py", line 303, in <module>
    main()
  File "../src/test/objectstore/allocsim/ops_scraper.py", line 250, in main
    process_info = process_osd(osd.name)
  File "../src/test/objectstore/allocsim/ops_scraper.py", line 155, in process_osd
    initiated_at = datetime.datetime.fromisoformat(op['initiated_at'])
AttributeError: type object 'datetime.datetime' has no attribute 'fromisoformat'